client/components/theme/index.jsx

```diff
@@ -61,11 +61,44 @@ const Theme = React.createClass( {
 		actionLabel: React.PropTypes.string
 	},
 
+	areButtonContentsEqual( nextButtonContents, buttonContents ) {
+		if ( ! isEqual( Object.keys( nextButtonContents ), Object.keys( buttonContents ) ) ) {
+			return false;
+		}
+
+		return every( buttonContents, ( option, name ) => {
```

Comment:
Although, the keys are the displayed items, right? So they may be the only parts that matter for the render. If indeed even they matter, since the button items aren't even displayed until click.